### PR TITLE
5501 make search optional

### PR DESCRIPTION
**Pull request by bradystroud**
**Summary**

issue #5501 

Adding `search: false` to config.yml will disable the search and filters which can sometimes cause the CMS to make too many API calls and hit the limit

**Test plan**

`yarn test` succeeded

**A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

File: packages/netlify-cms-core/src/components/Collection/CollectionControls.js [hunk 1]
```diff
@@ -48,8 +49,13 @@ function CollectionControls({
           filter={filter}
         />
       )}
-      {sortableFields.length > 0 && (
-        <SortControl fields={sortableFields} sort={sort} onSortClick={onSortClick} />
+      {isSearchEnabled && sortableFields.length > 0 && (
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/components/Collection/CollectionControls.js:52]: I would prefer not to mix these two concepts (sort and search) and you can already disable sorting via `sortable_fields: []`

File: packages/netlify-cms-core/src/components/Collection/__tests__/Sidebar.spec.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ jest.mock('../../../actions/collections');
 describe('Sidebar', () => {
   const props = {
     searchTerm: '',
+    isSearchEnabled: true,
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/components/Collection/__tests__/Sidebar.spec.js:23]: How about adding a new test with `isSearchEnabled: false`?

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @bradystroud, looking good per 2 code comments.

Also, can you add it to the docs here:
https://www.netlifycms.org/docs/configuration-options/#show-preview-links

between `Show Preview Links` and `Slug Type` with a description why would someone want to disable the search.

The docs file is here https://github.com/netlify/netlify-cms/blob/879e03b6f16feb23b49ba4248f2ad1eb3da4d259/website/content/docs/configuration-options.md#show-preview-links

File: packages/netlify-cms-core/src/components/Collection/CollectionControls.js (hunk 1 above)
Comment (bradystroud) [reply packages/netlify-cms-core/src/components/Collection/CollectionControls.js]: Hi @erezrokah,

I have removed the conditional rendering of the sorting button based on `isSearchEnabled`

Thanks,

-Brady

File: packages/netlify-cms-core/src/components/Collection/__tests__/Sidebar.spec.js (hunk 2 above)
Comment (bradystroud) [reply packages/netlify-cms-core/src/components/Collection/__tests__/Sidebar.spec.js:23]: Done in https://github.com/netlify/netlify-cms/pull/5509/commits/bc25bde9d105799834efa32531a0aded167b8577

**Review by erezrokah — APPROVED** [review]
Thanks @bradystroud, this is 🚀 

Great follow up on the review comments!

**Pull request merged**